### ISSUE TITLE
Overload implementation consistency check false positive

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2407]
When an overload function contains a default argument value, the implementation of that overload should also provide a default argument value. If it doesn't, this should be flagged as an error.

See this bug in the mypy issue tracker: https://github.com/python/mypy/issues/11308.

```python
from typing import overload, AnyStr, Any

@overload
def thing(path: None = ...) -> str:
    ...
@overload
def thing(path: AnyStr) -> str:
    ...
def thing(path: Any) -> str:
    return type(path).__name__

print(thing(None))
print(thing("wombat"))
print(thing(b"wombat"))
```
